factory/.env

```diff
@@ -0,0 +1,37 @@
+## This env represents the latest versions cyrpess supports. If you wish to push an older docker image,
+## please modify circle ci to release a one off branch instead of merging old versions to this file in master.
+
+# The debian image the factory is based on
+BASE_IMAGE='debian:bullseye-slim'
+
+# Node Versions: https://nodejs.org/en/download/releases/
+NODE_VERSION='16.18.1'
+
+# Update this to deploy the docker factory if you make changes to factory.Dockerfile or install scripts
+FACTORY_VERSION='1.0.0'
+
+# Chrome versions: https://www.ubuntuupdates.org/package/google_chrome/stable/main/base/google-chrome-stable
+CHROME_VERSION='107.0.5304.121-1'
+
+# Cypress versions: https://www.npmjs.com/package/cypress
+CYPRESS_VERSION='12.2.0'
+
+# Edge versions: https://packages.microsoft.com/repos/edge/pool/main/m/microsoft-edge-stable/
+EDGE_VERSION='100.0.1185.29-1'
+
+# Firefox versions: https://download-installer.cdn.mozilla.net/pub/firefox/releases/
+FIREFOX_VERSION='107.0'
+
+# Yarn versions: https://www.npmjs.com/package/yarn
+YARN_VERSION='1.22.19'
+
+# Webkit versions: https://www.npmjs.com/package/playwright-webkit
+# WEBKIT_VERSION='1.29.0'
+
+
+# Tags used for the docker images generated from the factory. Keep in mind the docker images will only release if these versions change.
+BASE_IMAGE_TAG="${NODE_VERSION}"
+
+BROWSERS_IMAGE_TAG="node-${NODE_VERSION}-chrome-${CHROME_VERSION}-ff-${FIREFOX_VERSION}-edge-${EDGE_VERSION}"
```

Comment:
is the idea that we already release the browsers image with chrome, firefox and edge? & once we provide webkit, we will also add the webkit version?

Confirming since we do have some images with just chrome & firefox. Assuming yes given the factory image can be used instead.

Reply:
Yes i'd expect the webkit version to be added to the list once we support it.

The tag version needs to indicate that something has updated. There (at least) two ways to do that, an arbitrary version(1.0.0) or a composite version (node-x-chrome-x). We're currently using the composite version which is nice because it encodes additional information into the version, ie what version of chrome is included. However with the composite version we need to add the versions of the things we care about or we can't easily release a new version.

For example if we used had the `node-16-chrome-107` version, added webkit support and pushed up the change, neither the version of chrome nor node have changed so the new version won't be pushed. So we have to change the version by adding webkit `node-16-chrome-107-webkit-10`, then the new version using webkit will be pushed.